Update globbing patterns and use SDK attribute

mlorbetske opened the pull request:
@phenning @seancpeters @davidfowl

Commit:
Update globbing patterns and use SDK attribute

dnfclas commented:
Hi __@mlorbetske__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

davidfowl commented:
Is the SDK thing just in prep for that change? Because it's incomplete, the props and targets are supposed to be gone as a result of the SDK node being there. Also the fact that the web templates globbing patterns got crazy makes me sad.

mlorbetske commented:
I'll remove the extra props & targets, sending you mail on what we're thinking post this upcoming release

Commit:
Remove extra props/targets imports

phenning commented on src/BaseTemplates/EmptyWeb/EmptyWeb.csproj:
```diff
@@ -9,8 +8,12 @@
   <ItemGroup>
-    <Compile Include="**\*.cs" />
-    <EmbeddedResource Include="**\*.resx" />
+    <Compile Include="**\*.cs" Exclude="$(GlobalExclude)" />
```
Just checking, the re-addition of all the Excludes of GlobalExclude is intentional, correct?

Commit:
Updated items to agree with SDK changes

Removed missing package reference from StarterWeb

Commit:
Use GlobalExclude items instead of the property

Exclude MiscFiles from the other items, exclude ContentExclude and
ContentFiles from None

phenning commented:
It looks like the removal of the Sdk.Web PackageReference isn't there for StarterWeb.csproj. Both WebAPI and Empty do show this as removed.

Commit:
Fix missed removal of Microsoft.Net.Sdk.Web

phenning commented:
Looks good :shipit:

Commit:
Move all globs in to csproj files

Improves perf over having the globs defined as they were in the props

phenning commented:
Other then this being really ugly, assuming it addresses the perf concerns, looks good from a template perspective.

:shipit:

mlorbetske commented:
@phenning ugly in a way we can change? Yes, that's to fix the perf issues

davidfowl commented on src/BaseTemplates/EmptyWeb/EmptyWeb.csproj:
```diff
@@ -8,10 +7,24 @@
+    <Content Include="**\*.json" Exclude="wwwroot\**\*.json" CopyToPublishDirectory="PreserveNewest" />
     <Folder Include="wwwroot\" />
+
+    <None Remove="@(Content);@(Compile);@(EmbeddedResource);$(GlobalExclude)" />
```
Does this part have to be in the project file too?

mlorbetske replied:
@davidfowl we can move the removes to the targets

mlorbetske replied:
https://github.com/aspnet/websdk/pull/69

phenning replied:
@mlorbetske I don't think so, at least for now.

mlorbetske replied:
@davidfowl the only issue with moving these to the targets is that CPS will no longer know where to put changes that override them, but this can be worked around if the user changes their csproj to remove the item from the item group & put it in to the right one before the targets run

Commit:
Remove removes, moved them to the targets file

phenning reviewed (commented):
This looks better without the removes.

mlorbetske commented:
Merging this to unblock broader testing, will submit another PR if needed

davidfowl commented:
@mlorbetske Do you know why the Reference items are needed for .NET Framework?
